### PR TITLE
Feature/iiwa 195

### PR DESCRIPTION
**Pull request by charkour**
Merge cost breakdown subtask into story.

### CODE REVIEW

File: src/backend-api/Controllers/ArchiveRecoverController.cs [hunk 1]
```diff
@@ -436,39 +436,40 @@ private void UpdateHardwareAssigning<T>(int employeeId)
...
+        /* ArchivedList<T>() returns a list of all archived items on a table.
+         */
+        public IActionResult ArchivedList<T>()
             where T : class, ISoftDeletable
-        {
-            var table = _context.Set<T>();
-            var list = table.Where(x => x.IsDeleted);
-            return Ok(list);
+        {   
+            return Ok(_context.Set<T>().Where(x => x.IsDeleted));
```
Comment (Xiscoder) [inline src/backend-api/Controllers/ArchiveRecoverController.cs:472]: nice

File: src/backend-api/Controllers/JobController.cs [hunk 2]
```diff
@@ -158,6 +180,48 @@ private MimeMessage CostBreakDownMessage(MimeMessage message)
...
+            DateTime today = DateTime.Today;
+            DateTime lastWeek = today.AddDays(-7);
+            decimal? costPrograms = _context.Set<T>()
+                    // Don't add any costs of deleted things.
+                    .Where(x => !x.IsDeleted)
+                    .Select(x => new
+                    {
+                        // Find the previous renewal date, and also select vars needed in future.
+                        PreviousRenewal = x.RenewalDate != null ?
+                        x.RenewalDate.Value.AddMonths(x.MonthsPerRenewal != null ? -x.MonthsPerRenewal.Value : -999) : new DateTime(1800),
```
Comment (Xiscoder) [inline src/backend-api/Controllers/JobController.cs:202]: Perhaps explain why we are setting previous renewal to 100 years ago if renewal date is null

**Review by Xiscoder — APPROVED** [review]

**Comment by Xiscoder** [comment]
Looks good!

**Comment by charkour** [comment]
Oops I will make the change to explain the 100 year old date in my next PR.

**Pull request merged**